prompting/mock.py

```diff
@@ -142,49 +289,64 @@ async def forward(
 
-            async def single_axon_response(i, axon):
+            async def single_axon_response(
+                i: int, target_axon: Union[bt.AxonInfo, bt.axon]
+            ):
                 """Queries a single axon for a response."""
 
-                t0 = time.time()
                 s = synapse.copy()
-                # Attach some more required data so it looks real
-                s = self.preprocess_synapse_for_request(axon, s, timeout)
-                # We just want to mock the response, so we'll just fill in some data
-                process_time = (
-                    random.random() * (self.max_time - self.min_time) + self.min_time
+
+                target_axon = (
+                    target_axon.info()
+                    if isinstance(target_axon, bt.axon)
+                    else target_axon
                 )
-                await asyncio.sleep(process_time)
-                if process_time < timeout:
-                    # Update the status code and status message of the dendrite to match the axon
-                    s.completion = f"Mock miner completion {i}"
-                    s.dendrite.status_code = 200
-                    s.dendrite.status_message = "OK"
-                else:
-                    s.completion = ""
-                    s.dendrite.status_code = 408
-                    s.dendrite.status_message = "Timeout"
 
-                s.dendrite.process_time = str(time.time() - t0)
+                # Attach some more required data so it looks real
+                s = self.preprocess_synapse_for_request(
+                    target_axon_info=target_axon, synapse=s, timeout=timeout
+                )
 
-                # Return the updated synapse object after deserializing if requested
-                if deserialize:
-                    return s.deserialize()
+                if is_stream:
+                    # If in streaming mode, return the async_generator
+                    return self.call_stream(
+                        synapse=s,
+                        timeout=timeout,
+                        deserialize=False,
+                    )
                 else:
-                    return s
-
+                    return await self.call(
+                        i=i,
+                        synapse=s,
+                        timeout=timeout,
+                        deserialize=deserialize,
+                    )
+
+            if not run_async:
+                return [
+                    await single_axon_response(target_axon) for target_axon in axons
+                ]
+
+            # If run_async flag is True, get responses concurrently using asyncio.gather().
             return await asyncio.gather(
                 *(
                     single_axon_response(i, target_axon)
                     for i, target_axon in enumerate(axons)
                 )
             )
 
-        return await query_all_axons(streaming)
+        return await query_all_axons(is_stream=streaming)
```

Comment:
Can we re-use any real dendrite code for this? It seems like we're getting closer to a full dendrite copy now with minor mocking

Reply:
This is a separate issue. Resolving